### PR TITLE
add connection callback on ping fail

### PR DESCRIPTION
**Pull request by danewalton**
# Checklist
- [ ] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-c/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [ ] This pull-request is submitted against the `master` branch. 
  - [ ] I have merged the latest `master` branch prior to submission and re-merged as needed after I took any feedback.
  - [ ] I have squashed my changes into one with a clear description of the change.

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem

# Description of the solution

### CODE REVIEW

File: iothub_client/src/iothubtransport_mqtt_common.c [hunk 1]
```diff
@@ -1878,6 +1878,7 @@ static void mqtt_error_callback(MQTT_CLIENT_HANDLE handle, MQTT_CLIENT_EVENT_ERR
             case MQTT_CLIENT_NO_PING_RESPONSE:
             {
                 LogError("Mqtt Ping Response was not encountered.  Reconnecting device...");
+                transport_data->transport_callbacks.connection_status_cb(IOTHUB_CLIENT_CONNECTION_UNAUTHENTICATED, IOTHUB_CLIENT_CONNECTION_COMMUNICATION_ERROR, transport_data->transport_ctx);
```
Comment (jebrando) [inline iothub_client/src/iothubtransport_mqtt_common.c]: >IOTHUB_CLIENT_CONNECTION_COMMUNICATION_ERROR [](start = 115, length = 44)

Should this be a new Status reason?  Maybe No Ping response?

Comment (danewalton) [reply iothub_client/src/iothubtransport_mqtt_common.c]: this seems reasonable. totally up to you

---
In reply to: [322858359](https://github.com/Azure/azure-iot-sdk-c/pull/1161#discussion_r322858359) [](ancestors = 322858359)

**Pull request merged**